front-end/src/pages/SettingsPage.tsx

```diff
@@ -0,0 +1,84 @@
+import axios from 'axios';
+import React, { useState } from 'react';
+import {
+    Card,
+    CardBody,
+    Button,
+    Form,
+    FormGroup,
+    Input,
+    Label,
+} from 'reactstrap';
+
+const SettingsPage = (loggedInUser: any) => {
```

Comment:
`loggedInUser` here can be `props`, since what we're passing in here is props

Bonus points if you can type the props parameter. An example of how to do this can be found here https://github.com/CMPUT404W21T02/group-project/blob/ba3760e021cff78cc8eb5944760e21d5885f8a3f/front-end/src/components/PostListItem.tsx#L6

Reply:
done. only thing is that now the settings page can be accessed by someone who isn't logged in (via `localhost:8000/settings`) but if they try to do anything, an error will always be displayed/alerted

Reply:
We can create something in React router that runs a conditional and does a redirect accordingly
https://reactrouter.com/web/example/auth-workflow.

I'm personally okay if we don't account for this edgecase, but if you'd like to tackle it at some point the link above should give some clues about where to start